# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/BlockReflowState.cpp
# Commit: b8a405d2e140
# Full Hash: b8a405d2e140de87b7a8b06e95a04f984a8ddf0a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-09-30 21:19:54
# Regressor Bug: 1731541
# File Overlap Count: 1
# Description:
#   Bug 1731541 - Implement text-wrap: balance for nsBlockFrame reflow. r=emilio
#   
#   A simple form of balance for short blocks, implemented by incrementally reducing
#   the effective inline-size used during line-breaking, up to the point where an
#   extra line would be created.
# ==============================================================================

diff -r 2a7646e1a4a5 -r b8a405d2e140 layout/generic/BlockReflowState.cpp
--- a/layout/generic/BlockReflowState.cpp	Sat Sep 30 15:53:11 2023 +0000
+++ b/layout/generic/BlockReflowState.cpp	Sat Sep 30 15:53:12 2023 +0000
@@ -27,17 +27,16 @@
 using namespace mozilla;
 using namespace mozilla::layout;
 
-BlockReflowState::BlockReflowState(const ReflowInput& aReflowInput,
-                                   nsPresContext* aPresContext,
-                                   nsBlockFrame* aFrame, bool aBStartMarginRoot,
-                                   bool aBEndMarginRoot,
-                                   bool aBlockNeedsFloatManager,
-                                   const nscoord aConsumedBSize,
-                                   const nscoord aEffectiveContentBoxBSize)
+BlockReflowState::BlockReflowState(
+    const ReflowInput& aReflowInput, nsPresContext* aPresContext,
+    nsBlockFrame* aFrame, bool aBStartMarginRoot, bool aBEndMarginRoot,
+    bool aBlockNeedsFloatManager, const nscoord aConsumedBSize,
+    const nscoord aEffectiveContentBoxBSize, const nscoord aInset)
     : mBlock(aFrame),
       mPresContext(aPresContext),
       mReflowInput(aReflowInput),
       mContentArea(aReflowInput.GetWritingMode()),
+      mInsetForBalance(aInset),
       mPushedFloats(nullptr),
       mOverflowTracker(nullptr),
       mBorderPadding(